Update introductory & concluding sections to match challenge-type spec. (#4)
peppelinux/draft-demarco-acme-openid-federation: draft-demarco-acme-openid-federation.md

--- abstract

The Automatic Certificate Management Environment (ACME) protocol [RFC8555]
allows server operators to obtain TLS certificates for their websites (HTTPS
[RFC2818]), based on a demonstration of control over the website's domain via a
fully-automated challenge/response protocol.

> OpenID Connect Federation 1.0 [OIDC-FED] is a standard that allows building

Reviewer: ```suggestion
OpenID Federation 1.0 is a standard that allows building
```

> OpenID Federation 1.0 defines how to build a trust infrastructure using trusted third party model.

Reviewer: ```suggestion
OpenID Federation 1.0 defines how to build a trust infrastructure
using a trusted third-party model.
```

> OpenID Federation 1.0 is a standard that allows building
> multilateral federations through a trust evaluation mechanism attesting the
> possession of public keys, signature capabilities, protocol specific metadata
> and several administrative and technical information in the form of trust marks,
> related to a specific entity belonging to an organization.

Reviewer: ```suggestion
OpenID Federation 1.0 defines how to build a trust infrastructure using trusted third party model.
It implements a trust evaluation mechanism attesting the
possession of public keys, protocol specific metadata
and several administrative and technical information in the form of trust marks,
related to a specific entity.
```

> This document defines how X.509 certificates associating a given OpenID Entity

Reviewer: ```suggestion
This document defines how X.509 certificates associating a given OpenID Federation Entity
```

> with a key included in that Entity's Configuration can be issued by a trust

Reviewer: Actually an ACME provider can be a leaf and not necessarly a Trust Anchor or an Intermediate entity.

anchor and its intermediates through the ACME protocol to all the organizations
that are part of a federation built on top of OpenID Connect Federation 1.0.

--- middle

# Introduction

> OpenID Connect Federation 1.0 allows an ACME server to issue X.509 certificates

Reviewer: ```suggestion
OpenID Federation 1.0 allows an ACME server to issue X.509 certificates
```

associating a given OpenID Entity to a key included in that Entity's

> configuration. Certificates can be provided to one or more organizations,

Reviewer: ```suggestion
Configuration. Certificates can be provided to one or more organizations,
```

without having pre-established any direct relationship or any stipulation of a
contract.

In a multilateral federation, composed by thousands of entities belonging to
different organizations, all the participants adhere to the same regulation or
trust framework. OpenID Connect Federation 1.0 allows each participant to
recognize the other participant using a trust evaluation mechanism, with RESTful
services and cryptographic materials.

Considering that a requestor is an entity requesting the issuance of a X.509

> certificate to a server and the issuer is the ACME server that validates the

Reviewer: ```suggestion
Certificate to a server and the issuer is the ACME server that validates the
```

entitlements of the requestor before issuing the X.509 certificate, this

> specification defines how ACME and OpenID Connect Federation 1.0 can be

Reviewer: ```suggestion
specification defines how ACME and OpenID Federation 1.0 can be
```

integrated to allow efficient issuance of X.509 certificates to a requestor via

> the introduction of a new ACME challenge type. The new challenge type extends

Reviewer: we must say in the terminolog ysection that "ACME Challenge is defined in RFC...." and therefore use it the defined term ACME Challenge as well as the other terms

the ACME protocol in the following ways:

- It associates a cryptographic key with an OpenID Entity, rather than a domain,
  since the authentication and authorization of the requestor is asserted with

>   OpenID Connect Federation 1.0.

Reviewer: ```suggestion
  OpenID Federation 1.0.
```

> - It defines how to use and validate a basic OpenID Connect Federation

Reviewer: ```suggestion
- It defines how to use and validate a basic OpenID Federation
```

  component, called Entity Configuration, that is a signed JWT published in a
  well-known resource (`/.well-known/openid-federation`) without requiring the
  `/.well-known/acme-challenge/{token}` endpoint.

- It defines how the OpenID Federation Entity Statements can be used for the

>   publication of the X.509 certificates, by a trust anchor or intermediate, that

Reviewer: ```suggestion
  publication of the X.509 certificates, by a Trust Anchor or Intermediate, that
```

>   publication of the X.509 certificates, by a trust anchor or intermediate, that

Reviewer 2: I think "trust anchor or intermediate" here is referring to the X.509 CA. I think we should pick some distinguished terminology for the X.509 CA, because "trust anchor" is used frequently in the context of OpenID Federation, so making the distinction clear would help with understanding.

on OpenID Connect Federation 1.0.

This specification can be implemented by:

- Federation Entities that join to a federation staging area using HTTP only
  transport to attest themselves as trustworthy, and then retrieve X.509
  certificates for their official HTTPS Federation Entity ID.

- Federation Entities that want to ask and obtain X.509 certificate for every
  Federation Key contained in their Entity Configuration, as made reliable in a
  Federation Trust Chain.

# Terminology

**ACME**:
: Automated Certificate Management Environment, a certificate management protocol [RFC8555].

**TA**:
: OpenID Connect Federation Trust Anchor, see CA.

**CA**:
: Certification Authority, also known as Trust Anchor or Intermediate,
  specifically one that implements the ACME protocol by serving an ACME server.

**CSR**:
: Certificate Signing Request, specifically a PKCS#10 [RFC2986] as supported by
  ACME.

**FQDN**:
: Fully Qualified Domain Name.

**Requestor**:
: Federation Entity that requests a X.509 certificate to a CA.

**Issuer**:
: Federation Entity that serves an ACME Server. The Federation Entity is then a
  CA.

> The terms "Federation Entity", "Trust Anchor", "Intermediate", "Entity

Reviewer: pelase move these above at the begin of the section and remove all the re-definition by using a short ref as mentioned before in one of the previous comments

Reviewer: Done

Configuration", "Entity Statement", "Trust Mark" and "Trust Chain" used in this
document are defined in the [Section
1.2](https://openid.net/specs/openid-connect-federation-1_0.html#name-terminology)
of [OIDC-FED].

# Conventions and Definitions

{::boilerplate bcp14-tagged}

# Certificates issued using OIDC Federation

The Issuer establishes the authorization of a Federation Entity to obtain
certificates for the identifier configured in the Requestor's Entity
Configuration.

The Federation Entity Keys are used to satisfy the Issuer's challenge, and the
public portion of the keys are included in the issued X.509 certificates.

The protocol assumes the following discovery preconditions are met. Then the
Issuer has the guarantee that:

1. The Requestor controls the private key related to the public part published

>    in its Entity Configuration, attested by the superior Entity Statement.

Reviewer: ```suggestion
   in its Entity Configuration, attested by the superior Subordinate Statement.
```

   discovery process to find which are the Issuers within the federation.

2. The Requestor requests and obtains a new nonce from the Issuer, by sending a
   HTTP HEAD request to the Issuer's `newNonce` resource;

3. The Issuer evaluates the trust to the Requestor by checking if it is part of
   the federation. If not the `newNonce` request MUST be rejected (**TBD** the
   error to return). There are two ways the Issuer is able to check if a
   Requestor is part of the federation, these are listed below:

    - The Requestor adds the Trust Chain JWS header parameter related to itself,
      this option is RECOMMENDED since it reduces the effort of the Issuer in
      evaluating the trust to the Requestor;

    - The Requestor doesn't add the Trust Chain in the request, then the Issuer
      MUST start a [Federation Entity
      Discovery](https://openid.net/specs/openid-connect-federation-1_0.html#section-8)
      to obtain the Trust Chain related to the Requestor.

4. The Requestor begins the certificate issuance process by sending a HTTP POST
   request to the Issuer's `newOrder` resource, and follows the remainder of the

>    ACME protocl as specified in [RFC8555], using the new challenge defined in

Reviewer: ```suggestion
   ACME protocol as specified in [RFC8555], using the new challenge defined in
```
